internal/configconverter/parser_provider.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config/configparser"
+	"go.opentelemetry.io/collector/service/parserprovider"
+)
+
+// converterProvider wraps a ParserProvider and accepts a list of functions that
+// convert ConfigMaps. The idea is for this type to conform to the open-closed
+// principle.
+type converterProvider struct {
```

Comment:
We may want to make use of https://github.com/open-telemetry/opentelemetry-collector/pull/40 when it's ready.